eam-light-backendweb/src/main/java/ch/cern/cmms/eamlightweb/meter/MeterRest.java

```diff
@@ -96,23 +98,24 @@ private List<MeterReadingWrap> getMeterEquipmentMeterReading(String equipmentCod
 		}
 
 		GridRequestResult gridResult = inforClient.getGridsService().executeQuery(authenticationTools.getR5InforContext(), gridRequest);
-
 		List<MeterReadingWrap> result = convertGridResultToObject(MeterReadingWrap.class,
 				null, gridResult
 		);
 
+		List<String> filteredUOM = result.stream().map(MeterReadingWrap::getUom).collect((Collectors.toList()));
+		GridRequest uomGridRequest = new GridRequest("BSUOMS", GridRequest.GRIDTYPE.LIST);
+		uomGridRequest.addFilter("uomcode", String.join(",", filteredUOM), "IN", GridRequestFilter.JOINER.OR);
+
+		List<Pair> uomDesc = convertGridResultToObject(Pair.class,
+				Pair.generateGridPairMap("uomcode", "uomdescription"),
+				inforClient.getGridsService().executeQuery(authenticationTools.getR5InforContext(), uomGridRequest));
+
+		Map<String, String> uomDescMap = uomDesc.stream()
+				.collect(Collectors.toMap(Pair::getCode, Pair::getDesc));
+
 		result.forEach(meter -> {
-			try {
 			meter.setEquipmentCode(equipmentCode);
-				GridRequest uomGridRequest = new GridRequest("BSUOMS", GridRequest.GRIDTYPE.LIST);
-				uomGridRequest.addFilter("uomcode", meter.getUom(), "=");
-			GridRequestResult uomDescResult = inforClient.getGridsService().executeQuery(authenticationTools.getR5InforContext(), uomGridRequest);
-			List<Pair> uomDesc = convertGridResultToObject(Pair.class,
-						Pair.generateGridPairMap("uomcode", "uomdescription"), uomDescResult);
-			meter.setUomDesc(uomDesc.get(0).getDesc());
-			} catch (InforException e) {
-				throw new RuntimeException(e);
-			}
+			if(meter.getUom()!=null) meter.setUomDesc(uomDescMap.get(meter.getUom()));

```

Comment:
we can filter the stream before. Also, if there is already a UomDesc we should not replace